### PR TITLE
Boturl update

### PR DESCRIPTION
**Pull request by pkakelas**
screenshot from 2014-03-04 22 54 35
screenshot from 2014-03-04 22 55 57
screenshot from 2014-03-04 22 56 16

### CODE REVIEW

File: controllers/bot.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                $error = str_replace( "initiate_", "", $error );
+                if ( strpos( $error, '_not_set' ) ) {
+                    $error = 'invalid_json_dictionary';
+                }
+                go( 'bot', 'update', [ 'bot_not_success' => true, 'error' => $error ] );
+            }
+            $user->save();
+            go( 'bot', 'update', [ 'bot_success' => true ] );
+        }
+        public function updateView( $boturl_empty, $boturl_invalid, $bot_success, $bot_not_success, $error ) {
```
Comment (dionyziz) [inline controllers/bot.php]: Where is `boturl_invalid` set?

File: views/bots/update.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?php
+    require 'views/header.php';?>
+
+    <p>To begin playing, you must set up your bot.<a href=''> Start by reading the tutorial.</a></p><?php
+    if ( $bot_success ) {
+        ?><p class='check'>Your bot is correctly configured <img src='static/images/check.png' alt='' /></p><?php
```
Comment (dionyziz) [inline views/bots/update.php]: Use proper alt texts.

File: views/bots/update.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+    elseif ( $bot_not_success ) {
+        ?><p class='error'>Your bot is incorrectly configured <img src='static/images/cancel.png' alt='' /></p><?php
```
Comment (dionyziz) [inline views/bots/update.php]: Use proper alt texts.

File: views/bots/update.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+    require 'views/header.php';?>
+
+    <p>To begin playing, you must set up your bot.<a href=''> Start by reading the tutorial.</a></p><?php
+    if ( $bot_success ) {
+        ?><p class='check'>Your bot is correctly configured <img src='static/images/check.png' alt='' /></p><?php
+    }
+    elseif ( $bot_not_success ) {
```
Comment (dionyziz) [inline views/bots/update.php]: nit: Separate the words `else` and `if`.

File: views/bots/update.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<?php
+    require 'views/header.php';?>
```
Comment (dionyziz) [inline views/bots/update.php]: nit: New line before `?>`

File: views/bots/update.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                ?><p class='error'>Your bot is running, but not responding to initiation. Did you write code to handle initiation?</p><?php
+                break;
+            case 'invalid_json':
+                ?><p class='error'>Your bot is not sending valid JSON. Did you write code to generate JSON correctly?</p><?php
+                break;
+            case 'invalid_json_dictionary':
+                ?><p class='error'>You must set the bot name, version, and your username. Did you build the correct JSON dictionary?</p><?php
+                break;
+            case 'username_mismatch':
+                ?><p class='error'>Your bot is not using your username. Did you set your username correctly?</p><?php
```
Comment (dionyziz) [inline views/bots/update.php]: Pull `<p class='error'>` and `</p>` out of switch. Consider using a dictionary:

```
$errors = [
    'could_not_resolve' => 'Your bot hostname is invalid. Did you enter a valid hostname?',
    ...
];
echo $errors[ $error ];
```

File: controllers/bot.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+            require_once 'models/grader/bot.php';
+            if ( empty( $boturl ) ) {
+                go( 'bot', 'update', [ 'boturl_empty' => true ] );
+            }
+            if ( !filter_var( $boturl, FILTER_VALIDATE_URL ) ) {
+                go( 'bot', 'update', [ 'boturl_invalid' => true ] );
+            }
+            if ( isset( $_SESSION[ 'user' ] ) ) {
+                $user = $_SESSION[ 'user' ];
+            }
```
Comment (dionyziz) [inline controllers/bot.php:14]: What if it's not set?

File: controllers/bot.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+            try {
+                $bot->sendInitiateRequest(); 
+            }
+            catch ( GraderBotException $e ) {
+                $error = end( $bot->errors );
+                $error = str_replace( "initiate_", "", $error );
+                if ( strpos( $error, '_not_set' ) ) {
+                    $error = 'invalid_json_dictionary';
+                }
+                go( 'bot', 'update', [ 'bot_not_success' => true, 'error' => $error ] );
```
Comment (dionyziz) [inline controllers/bot.php]: :%s/bot_not_success/bot_fail?

File: views/bots/update.php [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        ?><p class='error'>Your bot is incorrectly configured <img src='static/images/cancel.png' alt='cross' /></p><?php
+        $errors = [
+            'could_not_resolve' => 'Your bot hostname is invalid. Did you enter a valid hostname?',
+            'could_not_connect' => 'Your bot is unreachable on the network. Did you enter your public IP address?',
+            'http_code_not_ok' => 'Your bot is running, but not responding to initiation. Did you write code to handle initiation?',
+            'invalid_json' => 'Your bot is not sending valid JSON. Did you write code to generate JSON correctly?',
+            'invalid_json_dictionary' => 'You must set the bot name, version, and your username. Did you build the correct JSON dictionary?',
+            'username_mismatch' => 'Your bot is not using your username. Did you set your username correctly?'
+        ];
+        ?><p class='error'><?php echo $errors[ $error ] ?></p><?php
```
Comment (dionyziz) [inline views/bots/update.php]: nit: new line after `<?php`, new line before `?>`

File: views/bots/update.php [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            'could_not_resolve' => 'Your bot hostname is invalid. Did you enter a valid hostname?',
+            'could_not_connect' => 'Your bot is unreachable on the network. Did you enter your public IP address?',
+            'http_code_not_ok' => 'Your bot is running, but not responding to initiation. Did you write code to handle initiation?',
+            'invalid_json' => 'Your bot is not sending valid JSON. Did you write code to generate JSON correctly?',
+            'invalid_json_dictionary' => 'You must set the bot name, version, and your username. Did you build the correct JSON dictionary?',
+            'username_mismatch' => 'Your bot is not using your username. Did you set your username correctly?'
+        ];
+        ?><p class='error'><?php
+            echo $errors[ $error ]; 
+        ?></p><?php
```
Comment (dionyziz) [inline views/bots/update.php]: Trailing whitespace

Comment (pkakelas) [inline views/bots/update.php]: I can't see it

Comment (dionyziz) [inline views/bots/update.php]: Open it in vim.

Comment (pkakelas) [inline views/bots/update.php]: I use vim

**Pull request merged**